chevah/keycert/ssh.py

```diff
@@ -690,6 +697,8 @@ def _fromString_PUBLIC_OPENSSH(cls, data):
         @raises BadKeyError: if the blob type is unknown.
         """
         blob = base64.decodestring(data.split()[1])
+        if cls._guessStringType(blob) != 'blob':
+            raise BadKeyError("Fail to parse OpenSSH key content.")
```

Comment:
If required, you can move the `cls._guessStringType(blob) != 'blob'` check in `_fromString_BLOB`.... but I hope that we can design the code with as few checks and if conditionals as possible